NYPL/MyLibraryNYCApp
# Tiger team feature

## JIRA tickets:
- [MyLibraryNYC Investigation(s)](https://newyorkpubliclibrary.atlassian.net/browse/TGR-112)
- [MyLibraryNYC: Convert 'local' environment to 'development'](https://newyorkpubliclibrary.atlassian.net/browse/TGR-118)
- [MyLibraryNYCApp: Document changes to local environment](https://newyorkpubliclibrary.atlassian.net/browse/TGR-128)
- [MyLibraryNYCApp: Implement GitHub Actions for deployments](https://newyorkpubliclibrary.atlassian.net/browse/TGR-127)
- [My Library NYC: Make unit tests run when application builds](https://newyorkpubliclibrary.atlassian.net/browse/TGR-130)
- [MyLibraryNYC: Package logstash into the application](https://newyorkpubliclibrary.atlassian.net/browse/TGR-116)



## Things Done:
- Upgraded Ruby (to 3.3), gem dependencies, and Node (to 20.x)
- Implemented Docker + docker compose for local development
- Renamed the `local` environment to `development` and deprecated code related to the vestigial Elastic Beanstalk development environment
- Replaced Travis CI with GitHub Actions
- Made unit tests run automatically on PRs against protected branches (qa / production)

## How to Test:
- The build builds and the unit tests pass
- `docker-compose build`
- `docker-compose up`
- The app behaves as expected at `http://localhost:3000`
